fdbserver/Coordination.actor.cpp

```diff
@@ -545,15 +545,33 @@ struct LeaderRegisterCollection {
 	}
 };
 
+// extract the prefix descriptor from cluster id
+StringRef getClusterDescriptor(Key key) {
+	StringRef str = key.contents();
+	return str.eat(":");
+}
+
 // leaderServer multiplexes multiple leaderRegisters onto a single LeaderElectionRegInterface,
 // creating and destroying them on demand.
-ACTOR Future<Void> leaderServer(LeaderElectionRegInterface interf, OnDemandStore* pStore, UID id) {
+ACTOR Future<Void> leaderServer(LeaderElectionRegInterface interf,
+                                OnDemandStore* pStore,
+                                UID id,
+                                Reference<ClusterConnectionFile> ccf) {
 	state LeaderRegisterCollection regs(pStore);
 	state ActorCollection forwarders(false);
 
 	wait(LeaderRegisterCollection::init(&regs));
 
 	loop choose {
+		when(CheckDescriptorMutable req = waitNext(interf.checkDescriptorMutable.getFuture())) {
+			Optional<LeaderInfo> forward = regs.getForward(req.key);
+			if (forward.present()) {
+				req.reply.send(CheckDescriptorMutableReply{ false });
```

Comment:
In the case that we have a forward, the client should probably retry against the new coordinators or report a different error. We may need a separate return value to indicate this different outcome.

Reply:
> the client should probably retry against the new coordinators or report a different error.

I can return a different error `coordinators_changed()` in this case